### DISCUSSION TITLE
Organizing icons

### DISCUSSION THREAD

**Discussion by pali6** [goonstation/goonstation#6661] [Future Development Planning]
It'd be a good idea to organize where dmi files are, what they are named, in which dmi files which icon states are and how those are named. This should be a place to hopefully discuss some basic ideas on how this stuff should be organized.

Some of the current general issues:

- many unused icon states
- non-effects in the icons/effects folder
- dmi files containing random unsorted junk (station_objs.dmi etc.)
- dmi files not being in folders according to their purpose
- some icon states pointlessly duplicated (whether across multiple files or in a single file)
- probably inconsistent names of icon states but that's not too big of an issue really

Once people come up with some clear subgoals of this we can make a GitHub Project and track progress there.

**Comment by Enderthunder** [comment]
as self proclaimed janitor and sorter i think this is a massive YES, sortation will make implomentation and issue fixing oh so much easier.

**Comment by Arthur-Holiday** [comment]
- Ability icons need to get organized jn some way, currently they are weirdly spread around 2 different folders.
- Turfs definitely need some organization too, maybe we could have something like station turfs, azone turfs and outside turfs.

**Comment by Gannets** [comment]
Many icons have been kept and suffixed with -old for historical preservation. This really isn't required and we should remove any icon that has since been superceded.

**Reply by Momoberry** [reply]
Or just have a specific part dedicated to outdated pieces, if people still want to have the history aspect in the code. But honestly,, just sort of pushing them off is probably for the best.

**Reply by Gannets** [reply]
Since everything is captured in the git commit history, I don't see the point in bloating the repo size with them. Move them to a separate repo entirely if they're really required.

**Reply by Momoberry** [reply]
Good point actually

**Comment by DimWhat** [comment]
Couple of things I wanna do myself:

- ~~Kill fat clothing FOREVER.~~ EDIT: PR merged

- ~~Organizing all human clothing dmis into a human_clothing file~~ EDIT: PRed!

- ~~Kill the frank reynolds clothing.~~ EDIT: PRs merged

- Maybe splitting head.dmi into several headwear dmis all bundled in a file Example: helmet.dmi, hat.dmi, space.dmi

- ~~rename the suit file to something better than "overcoats"~~ EDIT: Unnecessary. 

- EDITED AFTERTHOUGHT: free backpack object sprites from storage and into their own file.

Something I want done:

- Please for the love of all things somebody code support for splitting monkey clothing into a format just like human clothing. It's gotten crowded in that merge conflict magnet.

**Reply by Enderthunder** [reply]
> Couple of things I wanna do myself:
> 
> * Kill fat clothing FOREVER.
> * Organizing all human clothing dmis into a human_clothing file
> * Kill the frank reynolds clothing.
> * Maybe splitting head.dmi into several headwear dmis all bundled in a file Example: helmet.dmi, hat.dmi, space.dmi
> * rename the suit file to something better than "overcoats"
> 
> Something I want done:
> 
> * Please for the love of all things somebody code support for splitting monkey clothing into a formant just like human clothing. It's gotten crowded in that merge conflict magnet.

the fact that all of this ONLY applies to the "mob" and "objects" section is a sheeeesh

**Comment by Momoberry** [comment]
The main thing for me is definitely having a better system of grouping the themes (like dim was talking about), and have a _?standardized?_ system for each major theme (Clothing, Mobs, UI,). Sub folders like "Cosmetic/Armour/Space/Effect/Gimmick/Protection" for all clothing items, depending on the primary function for each clothing item. Example, Armoured vest is primarily for armour, so it gets plopped into armour, Engineering suit is primarily for space travel, which puts it into Space, Cluwne Mask is placed into gimmick, mech boots go into effect, etc etc. Although effect and gimmick could probably just be combined, but I was thinking of effect being specifically positives. There's also the fact you'd have to separate the hazards armour, and split them between armour/space/protection tho. 

Some items have multiple functions, but you'd still be able to narrow it down pretty easily like this. Like Biosuit would be in protection, but armoured biosuit would be in armour. This would actually probably add more folders in the grand scheme though. Overcoats has a similar theme going on and basically I just think it should be stretched across all of the clothing items for uniformity, as having an organization pattern persistent throughout it all would help a lot with the clutter. UI/HUD files having their own area in Mob, etc.

Just organizing a lot of it into the "Main" theme of where things should go would go a long way, using the primary theme of the file. Like the mob worn clothing should still be in clothing, in a folder for "Active Wearing" or something instead of all over mob, because it's still clothing.

**Comment by Flaborized** [comment]
I was asked for thoughts so my thoughts:

I think it would be particularly useful to have more granular use of folders. Right now HUD icons and mob icons and mob overlays and ect are all grouped into 'mob' and I think they could benefit from some subcategories. icons/mob/ability_buttons or icons/mob/mutantrace_bases are some easy examples. Could apply to obj too and already does to some degree, but the implementation is inconsistent. I think the aim of an organization-ing should be to make finding sprites *and* placing sprites in the right place easier, and  folders are my best idea for it.
